Description: Fix spurious failure on Xen partition devices without disk devices
 Zero out grub_errno in the case where we handle GRUB_ERR_UNKNOWN_DEVICE by
 falling back to the partition device, otherwise a later call to this
 function may fail spuriously.
Author: Colin Watson <cjwatson@ubuntu.com>
Origin: upstream, http://bazaar.launchpad.net/~vcs-imports/grub/grub2-bzr/revision/5092
Bug-Debian: http://bugs.debian.org/708614
Applied-Upstream: http://bazaar.launchpad.net/~vcs-imports/grub/grub2-bzr/revision/5092
Last-Update: 2013-08-13

Index: b/util/getroot.c
===================================================================
--- a/util/getroot.c
+++ b/util/getroot.c
@@ -2316,6 +2316,7 @@
 	    grub_util_warn
 	      (_("disk does not exist, so falling back to partition device %s"),
 	       os_dev);
+	    grub_errno = GRUB_ERR_NONE;
 
 	    canon = canonicalize_file_name (os_dev);
 	    drive = find_system_device (canon ? : os_dev, &st, 0, 1);
